File: source/javaclient/com/ib/client/ContractDescription.java
/* Copyright (C) 2013 Interactive Brokers LLC. All rights reserved.  This code is subject to the terms
 * and conditions of the IB API Non-Commercial License or the IB API Commercial License, as applicable. */

package com.ib.client;

import java.util.Arrays;

public class ContractDescription {
    private Contract m_contract;
    private String[] m_derivativeSecTypes;

    // Get
    public Contract contract() { return m_contract; }
    public String[] derivativeSecTypes() { return m_derivativeSecTypes; }

    // Set
    public void contract(Contract contract) { m_contract = contract; }
    public void derivativeSecTypes(String[] derivativeSecTypes) { m_derivativeSecTypes = derivativeSecTypes; }

    public ContractDescription() {
        m_contract = new Contract();
    }

    public ContractDescription( Contract p_contract, String[] p_derivativeSecTypes) {
        m_contract = p_contract;
        m_derivativeSecTypes = p_derivativeSecTypes;
    }

    @Override public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append( "conid: " + m_contract.conid() + "\n");
        sb.append( "symbol: " + m_contract.symbol() + "\n");
        sb.append( "secType: " + m_contract.secType().toString() + "\n");
        sb.append( "primaryExch: " + m_contract.primaryExch() + "\n");
        sb.append( "currency: " + m_contract.currency() + "\n");
        sb.append( "derivativeSecTypes: " + Arrays.toString(m_derivativeSecTypes) + "\n");
        return sb.toString();
    }
}

File: source/csharpclient/client/ContractDescription.cs
/* Copyright (C) 2013 Interactive Brokers LLC. All rights reserved.  This code is subject to the terms
 * and conditions of the IB API Non-Commercial License or the IB API Commercial License, as applicable. */

using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.InteropServices;
using System.Text;

namespace IBApi
{
    /**
     * @class ContractDescription
     * @brief contract data and list of derivative security types
     * @sa Contract, EClient::reqMatchingSymbols, EWrapper::symbolSamples
     */
    public class ContractDescription
    {
        private Contract contract;
        private string[] derivativeSecTypes;

         /**
         * @brief A contract data
         */
        public Contract Contract
        {
            get { return contract; }
            set { contract = value; }
        }

        /**
         * @brief A list of derivative security types
         */
        public string[] DerivativeSecTypes
        {
            get { return derivativeSecTypes; }
            set { derivativeSecTypes = value; }
        }

        public ContractDescription()
        {
            Contract = new Contract();
        }

        public ContractDescription(Contract contract, string[] derivativeSecTypes)
        {
            Contract = contract;
            DerivativeSecTypes = derivativeSecTypes;
        }

        public override string ToString()
        {
            return Contract.ToString() + " derivativeSecTypes [" + String.Join(", ", DerivativeSecTypes) + "]";
        }
    }
}
